### PR TITLE
New test framework

### PR DESCRIPTION
**Pull request by aditim1359**

### CODE REVIEW

**Comment by aditim1359** [comment]
Failing test_Test.py with sqlalchemy error, passing locally with the docker db set up.

File: mltrace/__init__.py [hunk 1]
```diff
@@ -36,6 +36,15 @@
     tests,
 )
 
+from mltrace.entities import (
```
Comment (shreyashankar) [inline mltrace/__init__.py:39]: This is included twice -- delete

File: mltrace/entities/base_component.py [hunk 2]
```diff
@@ -51,32 +51,30 @@ def afterRun(self, **local_vars):
...
         def my_function(arg1, arg2):
                 do_something()
-
             arg1 and arg2 are the arguments passed to the
             beforeRun and afterRun methods.
             We first execute the beforeRun method, then the function itself,
             then the afterRun method with the values of the args at the
             end of the function.
-
         ADD DESCRIPTION HERE ABOUT INPUT VARIABLEs and what they are
```
Comment (shreyashankar) [inline mltrace/entities/base_component.py:78]: Can you add the descriptions here?

File: mltrace/entities/components.py [hunk 3]
```diff
@@ -10,13 +10,13 @@
 
 class PreprocessingComponent(Component):
```
Comment (shreyashankar) [inline mltrace/entities/components.py:11]: Let's delete this file as the `Preprocessing` component is never actually used / we haven't started working on the library yet

File: mltrace/entities/tests.py [hunk 4]
```diff
@@ -17,11 +17,8 @@ def testZscore(self, df=pd.DataFrame, stdev_cutoff: float = 5.0):
         Checks to make sure there are no outliers using z score cutoff.
         """
         z_scores = (
```
Comment (shreyashankar) [inline mltrace/entities/tests.py:19]: Let's delete this file as the custom tests are never actually used / we haven't started working on the library yet

File: tests/test_component.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
+import unittest
+from mltrace.entities import components, base_component
```
Comment (shreyashankar) [inline tests/test_component.py:2]: remove import to components since we are not actually having the component library yet

**Review by shreyashankar — CHANGES_REQUESTED** [review]
Just added some nits -- will look into the failing tests in a bit!

File: mltrace/__init__.py (hunk 1 above)
Comment (aditim1359) [reply mltrace/__init__.py:39]: done

File: mltrace/entities/base_component.py (hunk 2 above)
Comment (aditim1359) [reply mltrace/entities/base_component.py:78]: updated

File: mltrace/entities/components.py (hunk 3 above)
Comment (aditim1359) [reply mltrace/entities/components.py:11]: removed

File: mltrace/entities/tests.py (hunk 4 above)
Comment (aditim1359) [reply mltrace/entities/tests.py:19]: removed

File: tests/test_component.py (hunk 5 above)
Comment (aditim1359) [reply tests/test_component.py:2]: removed

**Comment by shreyashankar** [comment]
FYI linter is still failing

**Review by shreyashankar — APPROVED** [review]
LGTM

**Pull request merged**